tembo-pgmq-python/tests/test_integration.py

```diff
@@ -193,6 +192,55 @@ def test_validate_queue_name(self):
             self.queue.validate_queue_name(invalid_queue_name)
         self.assertIn("queue name is too long", str(context.exception))
 
+    def test_transaction_create_queue(self):
+        """Test creating a queue within a transaction."""
+        try:
+            self.queue.create_queue("test_queue_txn", perform_transaction=True)
+            raise Exception("Intentional failure")
+        except Exception:
+            pass
+        finally:
+            queues = self.queue.list_queues(perform_transaction=False)
+            self.assertNotIn("test_queue_txn", queues)
```

Comment:
@tavallaie , I think during the transactional flow, we basically want users to control when transactions are opened and closed. In these test cases then I think I would expect there to be like a `.begin()` and a `.commit()` that are called before/after (out side of pgmq) the pgmq calls.

I'll need to look up the exact syntax in psycopg3, but in pseudo code, something like this:

```python
tx = connection.begin()
queue.create('somequeue', tx)
tx.execute('insert into some other table")
queue.send('somequeue', 'message', tx)
tx.commit()
```

I don't see a commit anywhere, so that might be why these tests are failing?

Reply:
thanks, I find my bug I didn't have any `.commit()` in my code.